pkg/controller/pkg/revision/reconciler.go

```diff
@@ -345,6 +355,18 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		return reconcile.Result{RequeueAfter: longWait}, errors.Wrap(r.client.Status().Update(ctx, pr), errUpdateStatus)
 	}
 
+	// Lint Package using generic package linter if version constraints apply.
+	if !pr.GetIgnoreCrossplaneConstraints() {
+		if err := r.generic.Lint(pkg); err != nil {
```

Comment:
This seems like a breakdown in the `generic` abstraction; we state that `r.generic` is a "generic" linter (it's not immediately clear to me how that's different from the existing linter", but in practice this logic appears to assume that the generic linter will be exactly `xpkg.PackageCrossplaneCompatible`.

Reply:
The existing linter has checks that are specific to a _type of package_ while the generic linter is for _all package types_. This enables us to gate checks on qualities that are applicable to all package types (in this case `ignoreCrossplaneConstraints`), which can only be determined at reconcile time (i.e. based on the specific PR we are reconciling, not the type of package).

Reply:
This still seems fishy to me. If I follow correctly what we're doing - running the generic linter - is gated on a more narrowly scoped condition - is ignore crossplane constraints not set. What would happen if someone configured the reconciler to use a different generic linter, that had nothing to do with Crossplane version constraints?

Reply:
@negz good point, I think we can just check this directly instead of using a linter :+1: